### PR TITLE
Increase informer re-sync value and reduction of configmap informer cache

### PR DESCRIPTION
**Pull request by sudswasavi**
Thi commit increases the value of informer re-sync period to 12 hours
This value was currently set to 30 seconds, which would mean that the
kubernetes API server would be polled too frequently for full sync.

This commit sets the value as used in the kube api server defaults:
https://github.com/kubernetes/kubernetes/blob/release-1.17/pkg/controller/apis/config/v1alpha1/defaults.go#L120

In addition to this, we also restrict the configmap informers to listen on specific namespaces for both advancedL4 and non-advancedL4. This brings down the indexer cache consumption significantly for systems that have huge number of configmaps.

### CODE REVIEW

**Review by monotosh-avi — APPROVED** [review]

**Comment by monotosh-avi** [comment]
build ako

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -147,14 +147,16 @@ func RandomSeq(n int) string {
...
 var informerInstance *Informers
 
+const informerDefaultResync = 12 * time.Hour
+
 func instantiateInformers(kubeClient KubeClientIntf, registeredInformers []string, ocs oshiftclientset.Interface, namespace string) *Informers {
 	cs := kubeClient.ClientSet
 	var kubeInformerFactory kubeinformers.SharedInformerFactory
 	if namespace == "" {
-		kubeInformerFactory = kubeinformers.NewSharedInformerFactoryWithOptions(cs, time.Second*30)
+		kubeInformerFactory = kubeinformers.NewSharedInformerFactoryWithOptions(cs, informerDefaultResync)
```
Comment (hemantavi) [inline pkg/utils/utils.go:156]: Can we please take the resync time as an argument to `instantiateInformers`? If no argument is provided, then we can take the default value of `informerDefaultResync`?

Comment (sudswasavi) [reply pkg/utils/utils.go:154]: Done.

**Comment by akshayhavile** [comment]
@sudswasavi : Out of context ask: can we take out this line https://github.com/sudswasavi/load-balancer-and-ingress-services-for-kubernetes-1/blob/sudswas-informer-sync/cmd/ako-main/main.go#L114? 

We are calling same function on line #116

**Comment by sudswasavi** [comment]
> @sudswasavi : Out of context ask: can we take out this line https://github.com/sudswasavi/load-balancer-and-ingress-services-for-kubernetes-1/blob/sudswas-informer-sync/cmd/ako-main/main.go#L114?
> 
> We are calling same function on line #116



> @sudswasavi : Out of context ask: can we take out this line https://github.com/sudswasavi/load-balancer-and-ingress-services-for-kubernetes-1/blob/sudswas-informer-sync/cmd/ako-main/main.go#L114?
> 
> We are calling same function on line #116

Nice catch. Done.

**Comment by hemantavi** [comment]
build ako

**Review by hemantavi — APPROVED** [review]

**Review by monotosh-avi — APPROVED** [review]

**Review by hemantavi — APPROVED** [review]

**Pull request merged**